riot/riot.py

```diff
@@ -556,19 +571,54 @@ def run(
                 assert command is not None
                 if cmdargs is not None:
                     command = command.format(cmdargs=(" ".join(cmdargs))).strip()
-                env_str = " ".join(f"{k}={v}" for k, v in env.items())
+                env_str = "\n".join(f"{k}={v}" for k, v in env.items())
                 logger.info(
-                    "Running command '%s' with environment '%s'.", command, env_str
+                    "Running command '%s' in venv '%s' with environment:\n%s.",
+                    command,
+                    venv_path,
+                    env_str,
                 )
+                # Copy over *nspkg.pth files so that we can handle namespace packages.
+                # This is strictly required by Python 2.7 and <3.3 (see PEP 420).
+                ns_files = []
                 try:
-                    # Pipe the command output directly to `out` since we
-                    # don't need to store it.
+                    sitepkgs = inst.site_packages_path
+                    assert sitepkgs is not None, inst
+                    for path in (_ for _ in inst.site_packages_list if _ != sitepkgs):
+                        if path:
+                            try:
+                                for ns in (
+                                    _
+                                    for _ in os.listdir(path)
+                                    if _.endswith("nspkg.pth")
+                                ):
+                                    ns_dest = os.path.join(sitepkgs, ns)
+                                    if os.path.isfile(ns_dest):
+                                        # child overrides parent
+                                        continue
+                                    ns_src = os.path.join(path, ns)
+                                    with open(ns_src) as ns_in:
+                                        # https://github.com/pypa/setuptools/blob/b62705a84ab599a2feff059ececd33800f364555/setuptools/namespaces.py#L44
+                                        content = ns_in.read().replace(
+                                            "sys._getframe(1).f_locals['sitedir']",
+                                            f"'{path}'",
+                                        )
```

Comment:
oh boy, this is gross

Reply:
very ☹️  but at least should be reliable 🤞  (and `sys._getframe(1).f_locals['sitedir']` isn't quite the nicest thing in the world either 😝 )